lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py

```diff
@@ -852,7 +852,11 @@ def move_directory_files( self, current_dir, source_dir, destination_dir ):
             os.makedirs(destination_directory)
         for dir_entry in os.listdir(source_directory):
             source_entry = os.path.join(source_directory, dir_entry)
-            if os.path.islink(source_entry):
+            if os.path.isdir(source_entry):
+                destination_subdir = os.path.join(destination_directory, dir_entry)
+                if not os.path.exists(destination_subdir):
+                    os.makedirs(destination_subdir)
+            elif os.path.islink(source_entry):
```

Comment:
Hmm...this worked in my testing and I was using a fairly complex install recipe that included lots of files and subdirectories that included more files  I'm not quite following your point on the problem.  Sorry, but can you clarify?  My eyes may not currently be seeing it correctly.  ;)